src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php

```diff
@@ -55,9 +55,9 @@ private function locateFilesInFilesInFilesDefinitions(array $locations) : Genera
         foreach ($locations as $location) {
             if (is_file($location)) {
                 yield $location;
+            } elseif (is_dir($location)) {
+                yield from $this->extractFilesFromDirectory($location);
             }
-
-            yield from $this->extractFilesFromDirectory($location);
```

Comment:
`is_dir()` returns `true` for a link pointing to a directory and `false` for a link pointing to a file. For `is_file()` it's the other way around. So as expected. (In case that wasn't obvious.)